src/JMS/Serializer/GenericDeserializationVisitor.php

```diff
@@ -161,6 +161,9 @@ public function startVisitingObject(ClassMetadata $metadata, $object, array $typ
     public function visitProperty(PropertyMetadata $metadata, $data, Context $context)
     {
         $name = $this->namingStrategy->translateName($metadata);
+        if ($this->hasAdvancedNamingStrategy()) {
```

Comment:
better do do `if ( instanceof ...) then "new strategy" else "old strategy" endif` instead of computing each time the `name` twice